### PR TITLE
Add back in the option to conf_remap the verify_server settings.

### PR DESCRIPTION
**Pull request by shinrich**
In PR #4414 I made changes to clean up the verify_server* actions.  As part of that PR I moved all the overridable options to ssl_server_name.  When I brought those changes back to our environment, Dave Carlin pointed out that we needed access to the path to determine whether we could enforce server verification yet.  In theory it could be done via the ssl_server_name file, but it was much more natural to be done via the remap rules.

However, in other cases, making exceptions for one or two origins, the ssl_server_name option makes much more sense.

This PR adds back in the logic to conf_remap override the verify_server* actions.  If action is overridden in both locations, the value in ssl_server_name will win.

As part of the PR, I added a test to exercise the conf_remap option.

### CODE REVIEW

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -4668,6 +4668,107 @@ HttpSM::send_origin_throttled_response()
...
+  bool set_backwards_compatible = false;
+  if (RecGetRecordSource("proxy.config.ssl.client.verify.server", &source, false) == REC_ERR_OKAY) {
+    if (txn_conf->ssl_client_verify_server || (source != REC_SOURCE_DEFAULT && source != REC_SOURCE_NULL)) {
+      // Assuming that conf override is using the old name only if the base
+      // config is.
+      int8_t verifyServer = 0;
+      if (txn_conf->ssl_client_verify_server) {
+        verifyServer = txn_conf->ssl_client_verify_server;
+      } else {
+        REC_EstablishStaticConfigByte(verifyServer, "proxy.config.ssl.client.verify.server");
```
Comment (SolidWallOfCode) [inline proxy/http/HttpSM.cc]: This is done potentially on every upstream connection? That seems somewhat risky.

File: doc/admin-guide/files/records.config.en.rst [hunk 2]
```diff
@@ -3489,11 +3492,14 @@ Client-Related Configuration
 
 .. ts:cv:: CONFIG proxy.config.ssl.client.verify.server.properties STRING ALL
    :reloadable:
+   :overridable:
 
    Configures |TS| for what the default verify callback should check during origin server verification.
 
    You can override this global setting on a per domain basis in the ssl_servername.yaml file using the :ref:`verify_server_properties attribute<override-verify-server-properties>`.
 
+   You can also override via the conf_remap plugin, but the changes in ssl_server_name.yaml will take precedence.
```
[the remark this answers is not in the document]
Comment (shinrich) [reply doc/admin-guide/files/records.config.en.rst]: It is kind of arbitrary.  You really shouldn't be using both features together.  So are long as we are deterministic and documented it shouldn't matter.

File: proxy/http/HttpSM.cc (hunk 1 above)
Comment (shinrich) [reply proxy/http/HttpSM.cc]: For performance concerns?  Or reload concerns?  We could move it into the HttpConfig object I suppose.

Comment (shinrich) [reply proxy/http/HttpSM.cc]: Alan clarified that I was using a call that read the value and set a callback.  Changed the call to just read the value.  No callback setting.

File: doc/admin-guide/files/records.config.en.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply doc/admin-guide/files/records.config.en.rst]: Hrrmmm. A bit inconsistent with standard practice, we may want to fix that later.

**Comment by shinrich** [comment]
Pushed changes to allow the per-transaction override version to take precedence.

Also removed verify.server (the original version) from being overridable.

**Pull request merged**